zephyr/include/sof/compiler_info.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright 2023 NXP
+ */
+
+#ifndef __SOF_COMPILER_INFO_H__
+#define __SOF_COMPILER_INFO_H__
```

Comment:
This is completely a SOF level construct, so I wonder if you could just add arm64 arch directory and add compiler_info.h there (e.g. sof/src/arch/arm64/include/arch/compiler_info.h).

I also looked at Zephyr. but they don't seem to export any common toolchain version string in toolchain.h. This just exports compiler capabilities in terms of preprocessor defines. So I don't think we can implement the SOF compiler_info interface using general Zephyr interface. Or did I miss something @dcpleung @nashif @marc-hb ?

Reply:
You're right, this is not arm64 specific. I have no problem with using the `arch/` hierarchy  (same goes for this one https://github.com/thesofproject/sof/pull/7458) but isn't our goal to get rid of it instead of adding more to it? Unless there's a better alternative to this (using Zephyr support) the only solution (apart from the `arch/` one) which I'm seeing ATM is to make this file a combination of all architecture-specific `compiler_info.h`. If this is not okay either then I guess there's no choice but to introduce `sof/src/arch/arm64/include/arch/compiler_info.h`.

Reply:
Should be a bit better now.